src/plugins/vis_builder/public/application/components/right_nav.tsx

```diff
@@ -76,25 +94,19 @@ export const RightNav = () => {
           cancelButtonText={i18n.translate('visBuilder.rightNav.changeVisType.cancelText', {
             defaultMessage: 'Cancel',
           })}
-          onCancel={() => setNewVisType(undefined)}
+          onCancel={() => setConfirmAggs(undefined)}
           onConfirm={() => {
-            dispatch(
-              setActiveVisualization({
-                name: newVisType,
-                style: types.get(newVisType)?.ui.containerConfig.style.defaults,
-                aggConfigParams: persistedAggParams,
-              })
-            );
+            dispatch(setActiveVisualization(confirmAggs));
 
-            setNewVisType(undefined);
+            setConfirmAggs(undefined);
           }}
           maxWidth="300px"
           data-test-subj="confirmVisChangeModal"
         >
           <p>
             <FormattedMessage
               id="visBuilder.rightNav.changeVisType.modalDescription"
-              defaultMessage="Changing the visualization type will reset all field selections. Do you want to continue?"
+              defaultMessage="Switching to this visualization type will result in the loss of some selected fields. Do you want to continue?"
```

Comment:
Yeah loss seems like a strong word. @KrooshalUX what do you suggest here?

Reply:
@ashwin-pc this is a question for docs/writing/editorial.

Reply:
It is possible, and the main reason i hesitated was to save time. Less edge cases to worry about given that the field names can sometimes be long and there are a lot of ways to skin that goat. e.g. what if the user uses the same field for the x and y axis, etc. Given the tight deadline and how rare this would be, i opted for the quick solution. We can also fast follow for this one.

Reply:
@joshuarrrr @ashwin-pc @KrooshalUX 

`defaultMessage="Switching to this visualization type causes the loss of certain field configurations. Do you want to continue?"`

As a user I'm left wondering what field configurations are lost. Would the user be able to identify them easily?

Reply:
@ashwin-pc Alternative message:

`Certain field configurations may be lost when changing visualization types and you may need to reconfigure those fields. Do you want to continue?` 

Please choose the message that is best of the targeted user. Both read clearly to me.